examples/algorithm_progression.py

```diff
@@ -0,0 +1,81 @@
+import algorithms
+import large_image
+import os
+import pathlib
+import sys
+import yaml
+
+
+def apply_algorithm(algorithm, source):
+    new_source = large_image.new()
+    for tile in source.tileIterator(
+        format=large_image.tilesource.TILE_FORMAT_NUMPY,
+        tile_size=dict(width=2048, height=2048),
+    ):
+        altered_data = algorithm(tile['tile'])
+        new_source.addTile(altered_data, tile['x'], tile['y'])
+    return new_source
+
+
+def save_result(source, iteration, output_folder):
+    filename = f'iteration_{iteration}.tiff'
+    filepath = pathlib.Path(output_folder, filename)
+    source.write(filepath, lossy=False)
```

Comment:
Sadly, I think `write` currently needs a string file path and not a pathlib filepath.